test/onnx/exporter/test_small_models_e2e.py

```diff
@@ -144,8 +147,243 @@ def forward(self, *x):
         onnx_program = torch.onnx.export(VisionModel(), args, dynamo=True)
         onnx_testing.assert_onnx_program(onnx_program)
 
+    def test_onnx_export_huggingface_llm_models_with_kv_cache(self):
+        model, kwargs, dynamic_axes, input_names, output_names = (
+            _prepare_llm_model_gptj_to_test()
+        )
+        onnx_program = torch.onnx.export(
+            model,
+            kwargs=kwargs,
+            input_names=input_names,
+            output_names=output_names,
+            dynamic_axes=dynamic_axes,
+            dynamo=True,
+        )
+        # TODO(titaiwang): Investigate why ORT fails without optimization
+        onnx_program.optimize()
+        onnx_testing.assert_onnx_program(onnx_program)
+
     # TODO(justinchuby): Test multi-output HOPs
 
 
+def _prepare_llm_model_gptj_to_test() -> (
+    tuple[
+        torch.nn.Module,
+        dict[str, Any],
+        dict[str, dict[int, str]],
+        list[str],
+        list[str],
+    ]
+):
+    model = GPTJForCausalLM.from_pretrained("hf-internal-testing/tiny-random-gptj")
+
+    input_ids = torch.tensor(
```

Comment:
nit: are these specific values critical? are can we replace by random ints in a given range (which will take up less space, and clarify things better).

Reply:
I was concerned about flaky, but we will see.